File: Tron-1.1.0/CppClient/Game/ai.cpp
#include "ai.h"

#include <iostream>
#include <vector>
#include <map>

#include "effolkronium/random.hpp"

using namespace std;
using Random = effolkronium::random_static;

using namespace koala::chillin::client;
using namespace ks::models;
using namespace ks::commands;


AI::AI(World *world): RealtimeAI<World*>(world)
{
}

AI::~AI()
{
}

void AI::initialize()
{
    cout << "initialize" << endl;
}

void AI::decide()
{
    cout << "decide" << endl;

    auto randomDirection = Random::get(
        {EDirection::Up, EDirection::Right, EDirection::Down, EDirection::Left}
    );
    changeDirection(randomDirection);

    if (this->world->agents()[this->mySide].wallBreakerCooldown() == 0)
        activateWallBreaker();
}


void AI::changeDirection(EDirection direction)
{
    ChangeDirection cmd;
    cmd.direction((ECommandDirection) direction);
    this->sendCommand(&cmd);
}

void AI::activateWallBreaker()
{
    ActivateWallBreaker cmd;
    this->sendCommand(&cmd);
}

File: Tron-1.1.0/CSharpClient/Game/AI.cs
using System;
using System.Collections.Generic;

using KoalaTeam.Chillin.Client;
using KS;
using KS.Commands;
using KS.Models;

using KSObject = KS.KSObject;

namespace Game
{
	public class AI : RealtimeAI<World, KSObject>
	{
		private readonly Random random = new Random();
		private int stage = 0;

		public AI(World world) : base(world)
		{
		}

		public override void Initialize()
		{
			Console.WriteLine("initialize");
		}

		public override void Decide()
		{
			Console.WriteLine("decide");

			ChangeDirection((EDirection)random.Next(Enum.GetNames(typeof(EDirection)).Length));
			if (this.World.Agents[this.MySide].WallBreakerCooldown == 0)
				ActivateWallBreaker();
		}


		public void ChangeDirection(EDirection direction)
		{
			this.SendCommand(new ChangeDirection() { Direction = (ECommandDirection?)direction });
		}

		public void ActivateWallBreaker()
		{
			this.SendCommand(new ActivateWallBreaker());
		}
	}
}
